client/verta/verta/operations/monitoring/alert/_alerter.py

```diff
@@ -16,41 +16,49 @@ class _Alerter(object):
 
     _TYPE = _AlertService.AlerterTypeEnum.UNKNOWN
 
-    def __init__(self, comparison):
-        if not isinstance(comparison, comparison_module._VertaComparison):
-            raise TypeError(
-                "`comparison` must be an object from verta.common.comparison,"
-                " not {}".format(type(comparison))
-            )
-
-        self._comparison = comparison
-
     def __repr__(self):
         return "<{} alert>".format(
             _AlertService.AlerterTypeEnum.AlerterType.Name(self._TYPE).lower()
         )
 
-    @property
-    def comparison(self):
-        return self._comparison
+    @classmethod
+    @abc.abstractmethod
+    def _get_proto_class(cls):
+        raise NotImplementedError
+
+    @classmethod
+    @abc.abstractmethod
+    def _get_alert_field(cls):
+        raise NotImplementedError
```

Comment:
Did you happen to test this in Python 2? Asking because I don't personally know if they can be chained like this, and I _vaguely_ remember hearing about issues with it before. *shakes fist angrily at Python 2*

Reply:
https://stackoverflow.com/questions/11217878/python-2-7-combine-abc-abstractmethod-and-classmethod

Ahhh yes.
I think this is also why I ended up just tacking `_TYPE` as class attributes rather than methods or properties.

Reply:
I've double checked and confirmed that the tests pass in Python 2.7

```
(verta-py2) daniel@Daniels-MacBook-Pro verta % pytest tests/operations/monitoring/alerts/test_alerter.py
============================================================================== test session starts ===============================================================================
platform darwin -- Python 2.7.18, pytest-4.6.11, py-1.10.0, pluggy-0.13.1
rootdir: /Users/daniel/workspace/modeldb/client/verta/tests, inifile: pytest.ini
plugins: hypothesis-4.57.1
collected 8 items

tests/operations/monitoring/alerts/test_alerter.py ........                                                                                                                [100%]

============================================================================ 8 passed in 3.17 seconds ============================================================================
(verta-py2) daniel@Daniels-MacBook-Pro verta % pytest tests/operations/monitoring/alerts/test_entities.py
============================================================================== test session starts ===============================================================================
platform darwin -- Python 2.7.18, pytest-4.6.11, py-1.10.0, pluggy-0.13.1
rootdir: /Users/daniel/workspace/modeldb/client/verta/tests, inifile: pytest.ini
plugins: hypothesis-4.57.1
collected 12 items

tests/operations/monitoring/alerts/test_entities.py ............                                                                                                           [100%]

================================================================================ warnings summary ================================================================================
operations/monitoring/alerts/test_entities.py::TestIntegration::test_add_notification_channels
operations/monitoring/alerts/test_entities.py::TestIntegration::test_set_status
operations/monitoring/alerts/test_entities.py::TestAlert::test_update_last_evaluated_at
operations/monitoring/alerts/test_entities.py::TestAlert::test_creation_query_params
operations/monitoring/alerts/test_entities.py::TestAlert::test_creation_override_datetimes
operations/monitoring/alerts/test_entities.py::TestNonReferenceAlerters::test_crud[alerter0]
operations/monitoring/alerts/test_entities.py::TestNonReferenceAlerters::test_crud[alerter1]
operations/monitoring/alerts/test_entities.py::TestNonReferenceAlerters::test_repr[alerter0]
operations/monitoring/alerts/test_entities.py::TestNonReferenceAlerters::test_repr[alerter1]
operations/monitoring/alerts/test_entities.py::TestReference::test_crud
operations/monitoring/alerts/test_entities.py::TestReference::test_repr
  /Users/daniel/workspace/modeldb/client/verta/verta/_internal_utils/_utils.py:84: DeprecationWarning: Using 'method_whitelist' with Retry is deprecated and will be removed in v2.0. Use 'allowed_methods' instead
    raise_on_status=False)  # return Response instead of raising after max retries

-- Docs: https://docs.pytest.org/en/latest/warnings.html
==================================================================== 12 passed, 11 warnings in 77.76 seconds =====================================================================
```